moduleframework/mtf_scheduler.py

```diff
@@ -0,0 +1,214 @@
+    def __init__(self, args, unknown):
+
+        self.module = os.environ['MODULE']
+        self.action = args.action
+
+        # its used for filepath in loadCli:
+        self.json_tmppath = None
+        site_libs = os.path.dirname(moduleframework.__file__)
+        mtf_tools = "tools"
+
+        # choose between TESTS and ADDITIONAL ENVIRONMENT from options
+        self.tests = ''
+        if args.linter:
+            self.tests = (
+                "{SITE_LIB}/{MTF_TOOLS}/*.py".format(SITE_LIB=site_libs, MTF_TOOLS=mtf_tools))
+        self.additionalAvocadoArg = ''
+
+        for param in unknown:
+            # take care of this, see tags for safe/unsafe:
+            # http://avocado-framework.readthedocs.io/en/52.0/WritingTests.html#categorizing-tests
+            if os.path.exists(param):
+                # this is list of tests in local file
+                self.tests += " {0} ".format(param)
+            else:
+                # this is additional avocado param
+                self.additionalAvocadoArg += " {0} ".format(param)
+
+        common.print_debug("tests = {0}".format(self.tests))
+        common.print_debug("additionalAvocadoArg = {0}".format(
+            self.additionalAvocadoArg))
+
+    def avocado_run(self):
+        self.json_tmppath = tempfile.mktemp()
+        avocado_args = "--json {JSON_LOG}".format(JSON_LOG=self.json_tmppath)
+        avocadoAction = "MODULE={MODULE} avocado {ACTION} {AVOCADO_ARGS}".format(
+            MODULE=self.module, ACTION=self.action, AVOCADO_ARGS=avocado_args)
+
+        # run avocado with right cmd arguments
+        bash = process.run("{AVOCADO} {a} {b}".format(
+            AVOCADO=avocadoAction, a=self.additionalAvocadoArg, b=self.tests), shell=True, ignore_status=True)
+        common.print_info(bash.stdout, bash.stderr)
+        return bash.exit_status
+
+    def avocado_general(self):
+        # additional parameters
+        # self.additionalAvocadoArg: its from cmd line, whats unknown to this tool
+        avocado_args = ""  # when needed => specify HERE your additional stuff
+        avocadoAction = "avocado {ACTION} {AVOCADO_ARGS}".format(
+            MODULE=self.module, ACTION=self.action, AVOCADO_ARGS=avocado_args)
+        bash = process.run("{AVOCADO} {a} {b}".format(
+            AVOCADO=avocadoAction, a=self.additionalAvocadoArg, b=self.tests), shell=True, ignore_status=True)
+        common.print_info(bash.stdout, bash.stderr)
+        return bash.exit_status
+
+    def show_error(self):
+        if os.path.exists(self.json_tmppath):
+            try:
+                # file has to by json, otherwise it fails
+                json_data = open(self.json_tmppath).read()
+                data = json.loads(json_data)
+            except (IOError, ValueError) as e:
+                # file is not readable as json: No JSON object could be decoded
+                print(e)
+                # fatal error when this command fails, its unexpected
+                exit(127)
```

Comment:
please remove this file `self.json_tmppath` if error appear causes that it let there this file, because of exit.

Reply:
ok, done